### ISSUE TITLE
CassandraLauncher does not replace outdated Cassandra bundle

### ISSUE THREAD

**Issue by TimMoore** [akka/akka-persistence-cassandra#327]
`CassandraLauncher` works by extracting a bundled Cassandra JAR from its resource path to a directory passed by the caller.

Currently, it [tests whether the `cassandra-bundle.jar` file already exists in that directory](https://github.com/akka/akka-persistence-cassandra/blob/0d8e507d59a9735de75eb421317df943d919a50a/cassandra-launcher/src/main/scala/akka/persistence/cassandra/testkit/CassandraLauncher.scala#L187) before extracting it, presumably as an optimization to prevent having to extract the same file on each use.

This means that if the `cassandra-bundle.jar` is updated in a new version of Akka Persistence Cassandra (as in v0.83 and v0.60), it will not overwrite the older Cassandra version automatically.

One solution would be to unconditionally extract the file, and live with the performance hit in the case where it hasn't changed.

Another possibility would be to make the Cassandra version available to `CassandraLauncher` somehow (e.g., using [`sbt-buildinfo`](https://github.com/sbt/sbt-buildinfo)) and put it in the extracted filename, as in `cassandra-bundle-3.11.2.jar`.

## Workaround

You can work around this by deleting the existing `cassandra-bundle.jar` manually.

**Comment by patriknw** [comment]
Is the default `cassandraDirectory` in target? Then a `clean` will remove it?

**Comment by TimMoore** [comment]
@patriknw yes, a `clean` works to remove it (although a bit overkill). It's just a bit surprising and unintuitive, but not a major problem.

I puzzled over this a bit myself when testing the v0.60 update in Lagom and had to dig through the source code a bit before I realized what was happening. So I think some users might run into it as well.

I can send a pull request when I have some spare time, but I think this isn't urgent.

**Comment by patriknw** [comment]
thanks
